File: QE-Selenium-Java/Activity12.java
package demos;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class Activity12 {

    public static void main(String[] args) {

        WebDriver driver = new FirefoxDriver();
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(15));

        driver.get("https://training-support.net/webelements/dynamic-content");

        System.out.println("Page Title: " + driver.getTitle());

        WebElement clickMeBtn = driver.findElement(
                By.xpath("//button[contains(text(),'Click me')]"));
        clickMeBtn.click();

        WebElement message = wait.until(
                ExpectedConditions.visibilityOfElementLocated(
                        By.xpath("//*[contains(text(),'release')]")));

        System.out.println("Message displayed: " + message.getText());

        driver.quit();
    }
}

File: QE-Selenium-Python/Activity12.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

driver = webdriver.Firefox()
wait = WebDriverWait(driver, 15)

driver.get("https://training-support.net/webelements/dynamic-content")

print("Page Title:", driver.title)

click_me_btn = driver.find_element(
    By.XPATH, "//button[contains(text(),'Click me')]"
)
click_me_btn.click()

message = wait.until(
    EC.visibility_of_element_located(
        (By.XPATH, "//*[contains(text(),'release')]")
    )
)

print("Message displayed:", message.text)

driver.quit()
